stable_diffusion_videos/stable_diffusion_walk.py

```diff
@@ -70,6 +70,58 @@ def make_video_ffmpeg(frame_dir, output_file_name='output.mp4', frame_filename="
     return video_path
 
 
+def resume(
+    output_dir="dreams",
+    name="berry_good_spaghetti",
+    disable_tqdm=False,
+    height=512,
+    width=512,
+    upsample=False,
+    fps=30,
+):
+    """Generate video frames/a video given a list of prompts and seeds.
```

Comment:
Decided against `resume` after writing this comment so you can probably ignore it

Reply:
Reasoning for `resume` function is pretty simple -- with `walk` we're passing a ton of parameters, or defaults, and it's unclear which is desired.

We can't tell if the user passed height=512, or height=768, or if that's the default, or if it's not.

If we're resuming and the user passed height, is it because it's missing or because they want to change it? Obviously that's not a good idea. What if they pass one that's different, and ask for resume? Do we honor it or silently drop it? Warn, or error?

What if they pass a different prompt array? Should we check for different prompts in the array? Use olds seeds, or expect different seeds? What if they pass seeds but no prompts? And about all of the other parameters -- warn or replace, if different? We also don't know if they're "different because non-default" or "different because manually passed as parameters that match defaults" -- or even "same as defaults but manually passed, where json indicates non-defaults" -- how do we even detect that? Also we then have to note which ones are different, calculate, replace the frames if they already exist, which adds extra logic.

We also have to know which changes to allow and which to ignore. Why allow for that giant mess of unsupported/undesired/undesirable behavior? If we're in `resume`, it's exactly that -- a resume. Nothing changes, so we don't allow them to pass other changes, we pull the data we want (the old data, saved in the json) and we use it. That's all we want to do with resume, so that's all it allows. If they know they need/want to change something, they can pull the json open and hack away. I only added the optional height/width/etc params to `resume` because old json doesn't have it -- you weren't writing those values before, and backward support is ideal.

I understand why you want to implement it into `walk`, but it creates a lot more changes, and a *lot* more potential for breakage, and I feel it's justified as its own stub-function. I'm willing to try to do it your way, but I very strongly believe a separate, enforced-proper function is the right call. It "just works" and it works well, with no significant changes to `walk`'s logic -- now or in the future -- other than adding low_vram to `resume`, and then forwarding it to `walk`.

Reply:
If anything I'd prefer to break backward compatibility and remove the extra parameters from `resume` and rely on json to include full details because pushing those params into either `resume` or a modified `walk` breaks things

Reply:
Also re; defaults vs supplied params, I thought about this. You could assign the variable as `=None` in the definition, then do a conditional assignment with defaults later, ie 

```
    t_height = Height if not Height=None else 512
    t_width = Width if not Width=None else 512
    latents_a = torch.randn(
        (1, pipeline.unet.in_channels, t_height // 8, t_width // 8),
        device=pipeline.device,
        generator=torch.Generator(device=pipeline.device).manual_seed(first_seed),
    )
```

This gives you the ability to run `walk` with a potential `Resume` flag, detect parameters as supplied (not None) and override, or as default (ie None) and assign them as either your internal default (512) or feed it from the json data later in code.

Man, trying to talk through code feels like word salad sometimes. Did you grok that?